### DISCUSSION TITLE
cameraroll dependency

### DISCUSSION THREAD

**Discussion by ZRunner** [GetStream/stream-chat-react-native#1525] [Q&A] [answered]
Hello there!

I noticed that this lib has "react-native-cameraroll" as a dependency, which seems to be a lib to access local photos in react native.
Unfortunately this lib is completely dead, there hasn't been a single commit since October 2021, and tickets and pull requests remain unanswered. This lack of activity also means that it forces the use of react 16 or 17, preventing the migration of many projects to react 18 released some time ago with all the improvements that come with it.

Is there any plan to get rid of this lib and replace it by another one? Or do the streamchat devs prefer to stay stuck in react 17? And is there a workaround to still switch to react 18 while this problem is solved?

Thanks a lot for your consideration!

**Comment by santhoshvai** [comment, marked as answer]
Hello @ZRunner ,

We have created a [fork](https://github.com/GetStream/react-native-cameraroll) of the "react-native-cameraroll" library  in our organization as it is not maintained anymore. and we strive to maintain the fork with bug fixes on top of the original repo. The fork does not force the use of react 16 or 17. We will switch to the fork in the v5 release of our react-native SDK as outlined [here](https://getstream.io/chat/docs/sdk/reactnative/v5/basics/upgrading-from-v4/#update-camera-roll-dependency).

A temporary workaround for v4 is to point to this fork in your app with the following entry in your `package.json`,
`"@react-native-community/cameraroll": "git+https://github.com/GetStream/react-native-cameraroll.git"`
